servant-server/CHANGELOG.md

```diff
@@ -3,6 +3,10 @@
 
 * Add `err422` Unprocessable Entity
   ([#646](https://github.com/haskell-servant/servant/pull/646))
+* Changed `HasServer` instances for `QueryParam` and `QueryParam` to throw 400
+  when parsing fails
```

Comment:
I would add some more alarmist note (maybe `IMPORTANT` or just `NOTICE`), since the change may break expectations of existing applications without a compilation error.